### PR TITLE
Spring RequestMapping support + unit tests

### PR DESCRIPTION
**Pull request by guifre**

### CODE REVIEW

File: src/main/java/com/jtmelton/asa/analysis/visitors/java/spring/RouteSpringAnnotationVisitor.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                String name = parameterVisitor.getName();
+
+                if (REQUEST_MAPPING_ANNOTATIONS.contains(type)) {
+                    if (isMethodLevelPath(ctx)) {
+                        methodLevelPaths.add(name);
+                    } else {
+                        classLevelPaths.add(name);
+                    }
+                }
+            }
```
Comment (ajsosa) [inline src/main/java/com/jtmelton/asa/analysis/visitors/java/spring/RouteSpringAnnotationVisitor.java:47]: Looks like you can simplify this to just one conditional check and preserve the logic.

```java
            if (REQUEST_MAPPING_ANNOTATIONS.contains(type)) {
                methods.add(type);

                RouteSpringParameterVisitor parameterVisitor = new RouteSpringParameterVisitor();
                ctx.getPayload().accept(parameterVisitor);
                String name = parameterVisitor.getName();

                if (isMethodLevelPath(ctx)) {
                    methodLevelPaths.add(name);
                } else {
                    classLevelPaths.add(name);
                }
            }
```

File: src/main/java/com/jtmelton/asa/analysis/visitors/java/spring/RouteSpringAnnotationVisitor.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public boolean isMethodLevelPath(AnnotationContext ctx) {
+        boolean isMethod = false;
+        ParserRuleContext parent = ctx.getParent();
+        while (parent.getParent() != null) {
+            if (parent instanceof ClassBodyDeclarationContext) {
+                ClassBodyDeclarationContext cBodyCtx = (ClassBodyDeclarationContext) parent;
+                if (cBodyCtx.memberDeclaration() != null)
+                    isMethod = true;
```
Comment (ajsosa) [inline src/main/java/com/jtmelton/asa/analysis/visitors/java/spring/RouteSpringAnnotationVisitor.java:60]: Small code style nitpick. Use curly braces for if statements. Optionally you could simplify. 

```java
isMethod = cBodyCtx.memberDeclaration() != null;
```

I think there exists some code in the project already that does this though ha ha. Feel free to fix them if you see it as you continue to add to the codebase :)

**Review by ajsosa — CHANGES_REQUESTED** [review]
It's been almost 2 years since I've looked at this project so I'm a bit rusty on it. But other than the two comments which are super minor, this looks good overall.

**Review by jtmelton — APPROVED** [review]
I'll make the minor updates requested after merge.

**Pull request merged**